app/forms/declaration/DeclarationAdditionalActors.scala

```diff
@@ -36,6 +38,13 @@ case class DeclarationAdditionalActors(eori: Option[Eori], partyType: Option[Str
 
 object DeclarationAdditionalActors extends DeclarationPage {
 
+  def buildObjectInstanceFromId(value: String): DeclarationAdditionalActors = {
+    val attributes: Array[String] = value.split('-')
+
+    if (attributes.length == 0) DeclarationAdditionalActors(None, None)
```

Comment:
Hmmm.  This looks fragile.  What if Eori or party type is None?
I'd advocate for using the `ListItem`  createId and findById methods.

Reply:
:thinking: I thought about that as well, but the logic makes both of them mandatory, although it does not protect us against future changes or DB migrations.

Let me have a look at that class

Reply:
See here for example of "findBy" - https://github.com/hmrc/customs-declare-exports-frontend/blob/bc47029d16178f2c7f6014392ea6f1e7e83ec26a/app/controllers/declaration/DocumentsProducedRemoveController.scala#L82

Reply:
And here how to create the Id on the summary page - https://github.com/hmrc/customs-declare-exports-frontend/blob/bc47029d16178f2c7f6014392ea6f1e7e83ec26a/app/views/declaration/documentsProduced/documents_produced.scala.html#L96